flaml/automl.py

```diff
@@ -2418,6 +2424,7 @@ def custom_metric(
 
         self._state.fit_kwargs = fit_kwargs
         custom_hp = custom_hp or self._settings.get("custom_hp")
+        self._skip_transform = skip_transform or self._settings.get("skip_transform")
```

Comment:
Since `skip_transform` is a boolean, please follow line 2387's logic to check whether it's None instead of following line 2426. Otherwise, when skip_transform is False, its value will be overridden by self._settings.

Reply:
Ok updated the logic and also added test